src/main/java/com/hotline/elements/Element.java

```diff
@@ -1,37 +1,89 @@
 package com.hotline.elements;
 
-import org.openqa.selenium.By;
-import org.openqa.selenium.WebElement;
+import com.hotline.webdriver.WebDriverFacade;
+import com.hotline.webdriver.WebDriverFactory;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOf;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
 
 public class Element {
-    private final By by;
+    private static final int LONG_WAIT_SECONDS = 20;
+    private static final int SHORT_WAIT_SECONDS = 4;
 
+    private final By by;
     private boolean waitBeforeUse;
     private Element parent;
     private WebElement webElement;
+    private WebDriverFacade driver;
 
-//    public Element(WebElement webElement) {
-//        this.by = null;
-//        this.waitBeforeUse = false;
-//        this.webElement = webElement;
-//    }
+    public WebDriverFacade getDriver() {
+        if (driver == null) {
+            return new WebDriverFacade(new WebDriverFactory());
+        } else {
+            return driver;
+        }
+    }
 
     public Element(By by) {
         this(by, true);
     }
 
-//    public Element(By by, Element parent) {
-//        this(by, parent, false);
-//    }
-
-//    public Element(By by, Element parent, boolean waitBeforeUse) {
-//        this.by = by;
-//        this.parent = parent;
-//        this.waitBeforeUse = waitBeforeUse;
-//    }
-
     public Element(By by, boolean waitBeforeUse) {
         this.by = by;
         this.waitBeforeUse = waitBeforeUse;
     }
+
+    public Element(WebElement webElement) {
+        this.by = null;
+        this.waitBeforeUse = false;
+        this.webElement = webElement;
+    }
+
+    protected boolean isDisplayed(int seconds) {
+        try {
+            waitUntilVisible(seconds);
+            return true;
+        } catch (TimeoutException | NoSuchElementException | StaleElementReferenceException ignore) {
+            return false;
+        }
+    }
+
+    public boolean isDisplayed() {
+        return isDisplayed(LONG_WAIT_SECONDS);
+    }
+
+    public WebElement getWebElement() {
```

Comment:
I don't really understand what's happening in this method

Reply:
@andyyarych If you are talking about line 54 I totally agree with you and deleted it